workers/unity/Packages/com.improbable.gdk.core/Worker/DefaultWorkerConnector.cs

```diff
@@ -14,58 +16,151 @@ public abstract class DefaultWorkerConnector : WorkerConnector
+                DefaultComponentVtable = new PassthroughComponentVtable()
+            };
         }
 
         protected override ReceptionistConfig GetReceptionistConfig(string workerType)
         {
-            ReceptionistConfig config;
-
             if (Application.isEditor)
             {
-                config = new ReceptionistConfig
+                return new ReceptionistConfig
                 {
-                    WorkerType = workerType,
-                    WorkerId = CreateNewWorkerId(workerType),
-                    UseExternalIp = UseExternalIp
+                    ReceptionistHost = RuntimeConfigDefaults.ReceptionistHost,
+                    ReceptionistPort = RuntimeConfigDefaults.ReceptionistPort,
+                    WorkerId = CreateNewWorkerId(workerType)
                 };
             }
-            else
-            {
-                var commandLineArguments = Environment.GetCommandLineArgs();
-                var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
-                config = ReceptionistConfig.CreateConnectionConfigFromCommandLine(commandLineArgs);
-                config.WorkerType = workerType;
-                config.UseExternalIp = UseExternalIp;
-                if (!commandLineArgs.ContainsKey(RuntimeConfigNames.WorkerId))
+
+            var commandLineArguments = Environment.GetCommandLineArgs();
+            var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
+            return new ReceptionistConfig
                 {
-                    config.WorkerId = CreateNewWorkerId(workerType);
-                }
+                    ReceptionistHost = CommandLineUtility.GetCommandLineValue(
+                        commandLineArgs, RuntimeConfigNames.ReceptionistHost, RuntimeConfigDefaults.ReceptionistHost),
+                    ReceptionistPort = CommandLineUtility.GetCommandLineValue(
+                        commandLineArgs, RuntimeConfigNames.ReceptionistPort, RuntimeConfigDefaults.ReceptionistPort),
+                    WorkerId = CommandLineUtility.GetCommandLineValue(
+                        commandLineArgs, RuntimeConfigNames.WorkerId, CreateNewWorkerId(workerType)),
+                };
+        }
+
+        protected override LocatorConfig GetLocatorConfig()
+        {
+            var commandLineArguments = Environment.GetCommandLineArgs();
+            var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
+
+            var loginToken = CommandLineUtility.GetCommandLineValue(
+                commandLineArgs, RuntimeConfigNames.LoginToken, string.Empty);
+            var steamDeploymentTag = CommandLineUtility.GetCommandLineValue(
+                commandLineArgs, RuntimeConfigNames.SteamDeploymentTag, string.Empty);
+            var steamTicket = CommandLineUtility.GetCommandLineValue(
+                commandLineArgs, RuntimeConfigNames.SteamTicket, string.Empty);
+
```

Comment:
Given we can get to this point and have all the above values as `string.Empty`, is it worth doing a small validation step? I.e. - can we warn the user before it hits the actual connection flow and errors there?

Reply:
I still think we should do _something_ here.

Reply:
missed that comment. Gonna add a simple check to ensure that either logintoken or steam tag & ticket are set

Reply:
added a check